lib/mapObjects/CArmedInstance.cpp

```diff
@@ -111,7 +111,7 @@ void CArmedInstance::updateMoraleBonusFromArmy()
 	{
 		b->val = 2 - static_cast<si32>(factionsInArmy);
 		description = boost::str(boost::format(VLC->generaltexth->arraytxt[114]) % factionsInArmy % b->val); //Troops of %d alignments %d
-		description = b->description.substr(0, description.size()-2);//trim value
+		description = description.substr(0, description.size()-3);//trim value
```

Comment:
What exactly is trimmed here?
Why 3 bytes/characters?

Reply:
It is simmilar in the upper one.
The arraytxt[114] is "\nTroops of %d alignments %d"
Which resolves to "\nTroops of 3 alignments -1"
I trim  " -1".
In the other place there is added " -1".
Maybe one should decide which aproach to use. I would recommend clearing arraytxt at start - while reading, but this would large change.